components/calendar/src/iso.rs

```diff
@@ -276,6 +277,52 @@ impl DateTime<Iso> {
             time: types::Time::try_new(hour, minute, second, 0)?,
         })
     }
+
+    // Minute count representation of calendars starting from 00:00:00 on Jan 1st, 1970.
+    pub fn minutes_since_local_unix_epoch(&self) -> i32 {
+        let minutes_a_hour = 60;
+        let hours_a_day = 24;
+        let minutes_a_day = minutes_a_hour * hours_a_day;
+        if let Ok(unix_epoch) = DateTime::new_iso_datetime(1970, 1, 1, 0, 0, 0) {
+            (Iso::fixed_from_iso(*self.date.inner())
+                - Iso::fixed_from_iso(*unix_epoch.date.inner()))
+                * minutes_a_day
+                + i32::from(self.time.hour.number()) * minutes_a_hour
+                + i32::from(self.time.minute.number())
+        } else {
+            0
+        }
+    }
+
+    // Convert minute count since 00:00:00 on Jan 1st, 1970 to ISO Date.
+    pub fn from_minutes_since_local_unix_epoch(
+        minute: i32,
+    ) -> Result<DateTime<Iso>, DateTimeError> {
+        let minutes_a_hour = 60;
+        let hours_a_day = 24;
+        let minutes_a_day = minutes_a_hour * hours_a_day;
+        let extra_days = minute / minutes_a_day;
+        if let Ok(unix_epoch) = DateTime::new_iso_datetime(1970, 1, 1, 0, 0, 0) {
+            let unix_epoch_days = Iso::fixed_from_iso(*unix_epoch.date.inner());
+            let date = Iso::iso_from_fixed(unix_epoch_days + extra_days);
+            match (
+                i8::try_from((minute / minutes_a_hour) % hours_a_day).ok(),
+                i8::try_from(minute % minutes_a_hour).ok(),
+            ) {
+                (Some(hour_field), Some(minute_field)) => DateTime::new_iso_datetime(
+                    date.year().number,
+                    date.month().ordinal as u8,
+                    date.day_of_month().0 as u8,
+                    hour_field as u8,
+                    minute_field as u8,
+                    0,
+                ),
+                _ => DateTime::new_iso_datetime(0, 0, 0, 0, 0, 0),
```

Comment:
I am not able to use `unwrap` and raise error here. The code here seems verbose.

Reply:
You should be able to use `unreachable!()`